src/sas/qtgui/Perspectives/Inversion/InversionPerspective.py

```diff
@@ -703,8 +703,15 @@ def removeData(self, data_list=None):
             # self.updateGuiValues()
             self.setupModel()
         else:
+            # Block the signal, otherwise self.displayChange, triggered by a change to self.dataList, will run in
+            #  parallel with self.updateGuiValues. If self.displayChange is not finished before self.updateGuiValues
+            #  is launched, plots that have been deleted may still get called, throwing AttributeErrors.
+            self.dataList.blockSignals(True)
             self.dataList.setCurrentIndex(0)
+            self.displayChange(0)
+            self.dataList.blockSignals(False)
             self.updateGuiValues()
+        self.enableButtons()
```

Comment:
`enableButtons()` is already called in `self.updateGuiValues`. Maybe move this `enableButtons` to after line 704, where it might be needed? Unless we can do the same signal blocking for the commented out `self.updateGuiValues()` call in line 703. It was probably commented out to fix the issue you are currently fixing properly.